monai/transforms/compose.py

```diff
@@ -168,3 +168,64 @@ def inverse(self, data):
         for t in reversed(invertible_transforms):
             data = apply_transform(t.inverse, data, self.map_items, self.unpack_items)
         return data
+
+
+class OneOf(Compose):
+    """
+    ``OneOf`` provides the ability to radomly choose one transform out of a
+    list of callables with predfined probabilities for each.
+
+    Args:
+        transforms: sequence of callables.
+        weights: probabilities corresponding to each callable in transforms.
+            Probabilities are normalized to sum to one.
+
+    OneOf inherits from Compose and uses args map_items and unpack_items in
+    the same way.
+    """
+
+    def __init__(
+        self,
+        transforms: Optional[Sequence[Callable]] = None,
+        weights: Optional[Sequence[float]] = None,
+        map_items: bool = True,
+        unpack_items: bool = False,
+    ) -> None:
+        if transforms is None:
+            transforms = []
+        self.transforms = ensure_tuple(transforms)
+        if weights is None:
+            if len(transforms) == 0:
+                weights = []
+            else:
+                weights = [1.0 / len(transforms)] * len(transforms)
+        if len(weights) != len(transforms):
+            raise AssertionError("transforms and weights should be same size if both specified as sequences.")
+        self.weights = ensure_tuple(self._normalize_probabilities(weights))
+        self.map_items = map_items
+        self.unpack_items = unpack_items
+        self.set_random_state(seed=get_seed())
```

Comment:
all of this sits in parent constructor

Reply:
Should be resolved in new commit. Thanks!